src/components/HistoricalForecast/HistoricalForecast.tsx

```diff
@@ -1,130 +1,17 @@
-import { useContext } from 'react';
-import { startOfYesterday } from 'date-fns';
-import { Formik, FormikHelpers, Form, Field } from 'formik';
-import LoadingButton from '@mui/lab/LoadingButton';
-import Grid from '@mui/material/Grid';
-import DateField from 'components/DateField';
-import LocationAutoCompleteField from 'components/LocationAutocompleteField';
-import SelectField from 'components/SelectField';
-import TEMPERATURE_UNITS_OPTIONS from 'constants/temperatureUnitsOptions';
-import SettingsContext from 'contexts/SettingsContext';
-import LanguageOption from 'types/languageOption';
-import LANGUAGE_OPTIONS from 'types/languageOptions';
-import SettingsContextType from 'types/settingsContextType';
-import TemperatureUnitOption from 'types/temperatureUnitOption';
-import VALIDATION_SCHEMA, { FormValuesType } from './validation';
-import * as classes from './styles';
-
-const LANGUAGE_CHOICE_LABEL_ID = 'language-label';
-const LANGUAGE_CHOICE_LABEL = 'Language';
-const LOCATION_AUTOCOMPLETE = 'location-select';
-const TEMPERATURE_UNITS_LABEL_ID = 'unit-label';
-const TEMPERATURE_UNITS_LABEL = 'Temperature unit';
+import { useState } from 'react';
+import HistoricalWeatherForm from 'components/HistoricalWeatherForm';
+import HistoricalWeatherWidget from 'components/HistoricalWeatherWidget';
+import { FormValuesType } from '../HistoricalWeatherForm/validation';
 
 export default function HistoricalForecast() {
-  const { onSelectLanguage, onSelectTemperatureUnit } =
-    useContext<SettingsContextType>(SettingsContext);
-  const INITIAL_FORM_VALUES = {
-    language: '',
-    startDate: null,
-    endDate: null,
-    location: null,
-    temperatureUnit: '',
-  };
-  const yesterday = startOfYesterday();
-
-  const handleSubmit = (
-    values: FormValuesType,
-    { resetForm }: FormikHelpers<FormValuesType>,
-  ) => {
-    onSelectLanguage(values.language);
-    onSelectTemperatureUnit(values.temperatureUnit);
-    console.log(values);
-    resetForm();
-  };
-
+  const [selectedSearchParams, onSelectSearchParams] =
```

Comment:
Let's simplify architecture of this page. You don't need a state with `formValues` that duplicates form state. Move submit handler to this component that will perform fetch that we put in `HistoricalWeatherWidget` and save result in a state with forecast here